base/missing.jl

```diff
@@ -107,18 +107,22 @@ max(::Missing, ::Any)     = missing
 max(::Any,     ::Missing) = missing
 
 # Rounding and related functions
-for f in (:(ceil), :(floor), :(round), :(trunc))
-    @eval begin
-        ($f)(::Missing; digits::Integer=0, base::Integer=0) = missing
-        ($f)(::Type{>:Missing}, ::Missing) = missing
-        ($f)(::Type{T}, ::Missing) where {T} =
-            throw(MissingException("cannot convert a missing value to type $T: use Union{$T, Missing} instead"))
-        ($f)(::Type{T}, x::Any) where {T>:Missing} = $f(nonmissingtype(T), x)
-        # to fix ambiguities
-        ($f)(::Type{T}, x::Rational) where {T>:Missing} = $f(nonmissingtype(T), x)
-        ($f)(::Type{T}, x::Rational{Bool}) where {T>:Missing} = $f(nonmissingtype(T), x)
-    end
-end
+round(::Missing, ::RoundingMode=RoundNearest; sigdigits::Integer=0, digits::Integer=0, base::Integer=0) = missing
+round(::Type{>:Missing}, ::Missing, ::RoundingMode=RoundNearest) = missing
+round(::Type{T}, ::Missing, ::RoundingMode=RoundNearest) where {T} =
+    throw(MissingException("cannot convert a missing value to type $T: use Union{$T, Missing} instead"))
+round(::Type{T}, x::Any, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+# to fix ambiguities
+round(::Type{T}, x::Rational, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+round(::Type{T}, x::Rational{Bool}, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+
+trunc(::Missing; kwargs...) = round(missing, RoundToZero; kwargs...)
+floor(::Missing; kwargs...) = round(missing, RoundDown; kwargs...)
+ ceil(::Missing; kwargs...) = round(missing, RoundUp; kwargs...)
+
+trunc(::Type{T}, ::Missing; kwargs...) where {T} = round(T, missing, RoundToZero; kwargs...)
```

Comment:
These don't correspond to existing methods.

Reply:
Fixed.